tests/jballerina-unit-test/src/test/resources/test-src/query/query-action.bal

```diff
@@ -862,6 +862,26 @@ function testJoinedQueryActionWithRegExp() {
     "(?:A|B|[ab-fgh]+(?im-x:[cdeg-k]??AAAB*[^abc-efg](?:A|B|[ab-fgh]+(?im-x:[cdeg-k]??A)|)|^|PQ?)|)|^|PQ?");
 }
 
+function testQueryActionWithCollectClauseInsideLeClause1() {
+    int[][] arr = [[1, 2], [3, 4], [5, 6]];
+    int total = 0;
+    from var i in arr
+        let int tot = from var j in i collect sum(j)
+        do {
+            total += tot;
+        };
+    assertEquality(21, total);
+}
+
+function testQueryActionWithCollectClauseInsideLeClause2() {
```

Comment:
The original issue intended when we passing float array as a functional argument, Did you check that scenario with tests?

Reply:
The problem resides within the parser, and the choice of the collection in the query is unrelated.